packages/api/src/utils/billing-types.ts

```diff
@@ -91,19 +91,27 @@ export interface BillingEnv {
   billing: BillingService
   customers: CustomersService
   subscriptions: SubscriptionsService
+  termsOfService: TermsOfServiceService
 }
 
 export type PaymentSettings = {
   paymentMethod: null | PaymentMethod
   subscription: W3PlatformSubscription
+  agreement?: Agreement
```

Comment:
Why not make it required? Maybe there is a use case for it, but I can't necessarily think of one. Could be nice to err on the side of making it explicit everywhere.

Reply:
When I got to the website I remembered the use case. If the end-user is only entering their credit card for later use but not necessarily as part of choosing a storage subscription they might get charged for, we don't always render the tos checkbox (I think?).

https://github.com/web3-storage/web3.storage/pull/2028/files#r998202155

So I'm in favor of leaving this optional insofar as we still need to accomodate that UI use case.

Reply:
I actually made it optional because the type was being reused on the `GET`, but that method probably shouldn't return the agreement (it might be preferable to have separate types).  But if we want to make it optional I should probably only `throw` when it's invalid (currently it'll throw if it's missing, so it's not truly optional).  I'll update.